elements/lisk-p2p/src/p2p.ts

```diff
@@ -641,6 +678,62 @@ export class P2P extends EventEmitter {
 		);
 	}
 
+	private _assignPeerKind(peerInfo: P2PPeerInfo): P2PPeerInfo {
```

Comment:
I'm thinking about asking the peerBook firstly to check if the peer already exists. Perhaps we don't even have to do anything else.

Reply:
I think this function is not necessary. If the peer doesn't exist in the `peerBook`, we should assign `PeerKind.NONE`. `this._sanitizedPeerLists` is in my opinion redundant if we add every peer to the peerBook at the beginning sanitizing them properly before. This includes adding their peer kinds.

Reply:
`this._sanitizedPeerLists` is only assigning the peerKind in the beginning, but lets say any peer that belongs to any peerKind get deleted from the peer list due to eviction and later when we receive it through others, it will be assigned `NONE`. So to ensure that we always assign the right peer kind to a peer, I have this check everytime we add a new peer.